### PR TITLE
Make currency delta transient

### PR DESCRIPTION
**Pull request by hensha256**

### CODE REVIEW

File: src/PoolManager.sol [hunk 1]
```diff
@@ -144,7 +145,7 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
     function _accountDelta(Currency currency, int128 delta) internal {
         if (delta == 0) return;
 
-        int256 current = currencyDelta[msg.sender][currency];
+        int256 current = msg.sender.getCurrencyDelta(currency);
```
Comment (snreynolds) [inline src/PoolManager.sol:148]: this syntax is kinda funky but I do like it as a library 🤷‍♀️

**Review by snreynolds — APPROVED** [review]

File: src/libraries/CurrencyDelta.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+
+library CurrencyDelta {
+    uint256 constant CURRENCY_DELTA_SLOT = uint256(keccak256("CurrencyDelta")) - 1;
+
+    function setCurrencyDelta(address locker, Currency currency, int256 delta) internal {
+        uint256 slot = CURRENCY_DELTA_SLOT;
+
```
Comment (snreynolds) [inline src/libraries/CurrencyDelta.sol:11]: actually could you potentially write a helper called hashKey = _computeKey(locker, currency)? this whole chunk is repeated in both set/get

**Review by snreynolds — CHANGES_REQUESTED** [review]

File: src/PoolManager.sol (hunk 1 above)
Comment (hensha256) [reply src/PoolManager.sol:148]: Yeah i wasnt 100% on it either. Would you prefer it as
`CurrencyDelta.get(msg.sender, currency)` and `CurrencyDelta.set(msg.sender, currency, delta)`
or
`msg.sender.getCurrencyDelta(currency)` and `msg.sender.setCurrencyDelta(currency, delta)`

File: src/libraries/CurrencyDelta.sol [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import {Currency} from "../types/Currency.sol";
+
+library CurrencyDelta {
+    uint256 constant CURRENCY_DELTA_SLOT = uint256(keccak256("CurrencyDelta")) - 1;
+
+    function _computeSlot(address locker, Currency currency) internal pure returns (bytes32 hashSlot) {
+        uint256 slot = CURRENCY_DELTA_SLOT;
+
+        assembly {
```
Comment (ewilz) [inline src/libraries/CurrencyDelta.sol:12]: why is this done in assembly? cheaper?

Comment (hensha256) [reply src/libraries/CurrencyDelta.sol:12]: Yeeah its quite a lot cheaper. And its like another 1000 cheaper if instead we do 
```
        assembly {
            mstore(0, locker)
            mstore(32, currency)
            hashSlot := keccak256(0, 64)
        }
```

with a single hash and not have a `CURRENCY_DELTA_SLOT` thats added into the process with a second hash. But I feel like its definitely best practice to add that just for if we ever had another mapping with these 2 parameters too. Thoughts?

File: src/PoolManager.sol (hunk 1 above)
Comment (hensha256) [reply src/PoolManager.sol]: took a poll and went with the first one!

File: src/libraries/CurrencyDelta.sol [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+
+library CurrencyDelta {
+    uint256 constant CURRENCY_DELTA_SLOT = uint256(keccak256("CurrencyDelta")) - 1;
+
+    function _computeSlot(address locker, Currency currency) internal pure returns (bytes32 hashSlot) {
```
Comment (dianakocsis) [inline src/libraries/CurrencyDelta.sol:9]: Do we still want this to be called locker?

Comment (hensha256) [reply src/libraries/CurrencyDelta.sol:9]: oh yeah good point

File: src/libraries/CurrencyDelta.sol (hunk 3 above)
Comment (snreynolds) [reply src/libraries/CurrencyDelta.sol:12]: This though is not standard for computing the mapping slot?

File: src/libraries/CurrencyDelta.sol [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+
+library CurrencyDelta {
+    uint256 constant CURRENCY_DELTA_SLOT = uint256(keccak256("CurrencyDelta")) - 1;
```
Comment (snreynolds) [inline src/libraries/CurrencyDelta.sol:7]: Instead of this computing a hash every time we can just use the hex val and comment above?

Comment (hensha256) [reply src/libraries/CurrencyDelta.sol:7]: oh yeeaaah nice haha, i made that PR after this

File: src/libraries/CurrencyDelta.sol (hunk 3 above)
Comment (hensha256) [reply src/libraries/CurrencyDelta.sol:13]: Yeah the standard way of computing a mapping is including a slot

**Review by snreynolds — APPROVED** [review]

**Pull request merged**